### PR TITLE
Release artifacts in maven central and formatting changes

### PR DESCRIPTION
**Pull request by narape**
# TL;DR
* Tell the release plugin to not stage the artifacts in maven central, but release it right away. 
* Configure spotless to format poms to not conflict with the release plugin.

## Type
 - [X] Bug Fix
 - [ ] Feature
 - [ ] Plugin

## Are all requirements met?

 - [ ] Code completed
 - [ ] Smoke tested
 - [ ] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
Currently artifacts are staged by maven central for manual inspection. And then doing `mvn nexus-staging:release` would release it. However I can verify it anyway, so lets release it and "test in production".

Also previous spotless config for poms had a conflict with the release:plugin. The release plugin changed from multipline `project` tag to a single line.
```diff
-<project xmlns="http://maven.apache.org/POM/4.0.0"
-  xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
-  xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
```
It is actually a [bug](https://issues.apache.org/jira/browse/MRELEASE-1008) in the release plugin. As no attachment to multiline `project` is held so this PR changes the format of spotless to produce single line `project` in poms.

## Tracking Issue
https://github.com/flyteorg/flyte/issues/2086

## Follow-up issue
_NA_

### CODE REVIEW

**Review by ckiosidis — APPROVED** [review]
lgtm just a bunch of refactoring
